src/rewriter/mod.rs

```diff
@@ -150,12 +217,24 @@ impl<'h, O: OutputSink> HtmlRewriter<'h, O> {
     ///
     /// [`OutputSink`]: trait.OutputSink.html
     pub fn new<'s>(settings: Settings<'h, 's>, output_sink: O) -> Self {
-        let encoding = settings.encoding;
+        let encoding = SharedEncoding::new(settings.encoding);
         let mut selectors_ast = selectors_vm::Ast::default();
         let mut dispatcher = ContentHandlersDispatcher::default();
-        let has_selectors = !settings.element_content_handlers.is_empty();
+        let has_selectors =
+            !settings.element_content_handlers.is_empty() || settings.adjust_charset_on_meta_tag;
+
+        let charset_adjust_handler = match settings.adjust_charset_on_meta_tag {
+            true => Some(handler_adjust_charset_on_meta_tag(SharedEncoding::clone(
```

Comment:
Just use `if .. else`

Reply:
I find matching on booleans to often be more readable than a `if`, specially when both expressions are similar (not the case here).

Happy to change it anyway.

Reply:
:heavy_check_mark: